### PR TITLE
INFRA-240: Update autorunner zip file name

### PR DESCRIPTION
**Pull request by chinzou**
https://mekomsolutions.atlassian.net/browse/INFRA-240

### CODE REVIEW

File: roles/usb_autorunner/files/autorunner.sh.j2 [hunk 1]
```diff
@@ -1,45 +1,48 @@
...
+SECRET_KEY_FILENAME=secret.key.enc
 
-USB_DATA_PATH=$1
+MOUNT_POINT=$1
 
-if [ ! -r $USB_DATA_PATH/$KEY_FILENAME ]
+if [ ! -r $MOUNT_POINT/$SECRET_KEY_FILENAME ]
 then
-  echo "key file is missing, exiting"
+  echo "Secret key file ‘$SECRET_KEY_FILENAME’ is missing at path $MOUNT_POINT/"
```
Comment (rbuisson) [inline roles/usb_autorunner/files/autorunner.sh.j2]: Looks like this is not exactly single quote but some other character.
Could you make sure those are " ' "

File: roles/usb_autorunner/files/autorunner.sh.j2 [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
-  decrypt_key_path=$(find $USB_DATA_PATH -maxdepth 1 -name $KEY_FILENAME)
+  autorun_file_path=$(find $MOUNT_POINT -maxdepth 1 -name $AUTORUN_FILENAME)
+  decrypt_key_path=$(find $MOUNT_POINT -maxdepth 1 -name $SECRET_KEY_FILENAME)
   
   # Clean workdir
   rm -f $AUTORUN_CONTENT_PATH/*
+  rm -rf $TEMP_PATH
+  mkdir -p $TEMP_PATH
+  
   # Decrypting and executing the autorun script
```
Comment (rbuisson) [inline roles/usb_autorunner/files/autorunner.sh.j2]: Why this space?

File: roles/usb_autorunner/files/autorunner.sh.j2 [hunk 3]
```diff
@@ -1,45 +1,48 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
+TEMP_PATH=/tmp/autorunner/
 AUTORUN_CONTENT_PATH={{ usb_autorunner_install_path }}/workdir
```
Comment (rbuisson) [inline roles/usb_autorunner/files/autorunner.sh.j2]: This would be better renamed: `AUTORUNNER_WORKDIR`

**Review by rbuisson — CHANGES_REQUESTED** [review]

File: roles/usb_autorunner/files/autorunner.sh.j2 [hunk 4]
```diff
@@ -1,45 +1,46 @@
...
-  rm -f $AUTORUN_CONTENT_PATH/*
+  rm -f $AUTORUNNER_WORKDIR/*
+  rm -rf $TEMP_PATH
+  mkdir -p $TEMP_PATH
+  
   # Decrypting and executing the autorun script
+  openssl rsautl -decrypt -oaep -inkey $PRIVATE_CERT -in $decrypt_key_path -out $AUTORUNNER_WORKDIR/secret.key
+  openssl enc -d -aes-256-cbc -pbkdf2 -in $autorun_file_path -out $AUTORUNNER_WORKDIR/autorun.zip -pass file:$AUTORUNNER_WORKDIR/secret.key
 
+  unzip -oq $AUTORUNNER_WORKDIR/autorun.zip -d $TEMP_PATH
```
Comment (rbuisson) [inline roles/usb_autorunner/files/autorunner.sh.j2]: This is the opposite we should do:
Decrypt on the tmp dir, and then unzip in the workdir.

File: roles/usb_autorunner/files/autorunner.sh.j2 [hunk 5]
```diff
@@ -1,45 +1,46 @@
...
   
   # Clean workdir
-  rm -f $AUTORUN_CONTENT_PATH/*
+  rm -f $AUTORUNNER_WORKDIR/*
+  rm -rf $TEMP_PATH
+  mkdir -p $TEMP_PATH
+  
   # Decrypting and executing the autorun script
+  openssl rsautl -decrypt -oaep -inkey $PRIVATE_CERT -in $decrypt_key_path -out $AUTORUNNER_WORKDIR/secret.key
+  openssl enc -d -aes-256-cbc -pbkdf2 -in $autorun_file_path -out $AUTORUNNER_WORKDIR/autorun.zip -pass file:$AUTORUNNER_WORKDIR/secret.key
```
Comment (rbuisson) [inline roles/usb_autorunner/files/autorunner.sh.j2]: Could you change this to:
```
openssl enc -d -aes-256-cbc -md sha256 -in $autorun_file_path -out $AUTORUNNER_WORKDIR/autorun.zip -pass file:$AUTORUNNER_WORKDIR/secret.key
```

Removing `-pbkdf2` as this is not supported in OpenSSL on MacOS :/
Adding `-md sha256`to ensure compatiblity between OpenSSL versions

**Review by rbuisson — CHANGES_REQUESTED** [review]

File: roles/usb_autorunner/tasks/main.yml [hunk 6]
```diff
@@ -72,7 +72,7 @@
   shell: openssl rsautl -encrypt -oaep -pubin -inkey {{ usb_autorunner_install_path }}/certificate/autorunner.pub.pem -in {{ temp_test_folder.path }}/secret.key -out {{ temp_test_folder.path }}/secret.key.enc
 
 - name: Encrypt the file
-  shell: openssl aes-256-cbc -pbkdf2 -in {{ temp_test_folder.path }}/startup.zip -out {{ temp_test_folder.path }}/startup.zip.enc -pass file:{{ temp_test_folder.path }}/secret.key
+  shell: openssl aes-256-cbc -pbkdf2 -in {{ temp_test_folder.path }}/autorun.zip -out {{ temp_test_folder.path }}/autorun.zip.enc -pass file:{{ temp_test_folder.path }}/secret.key
```
Comment (rbuisson) [inline roles/usb_autorunner/tasks/main.yml]: You might want to change this to:
```
openssl enc -aes-256-cbc -md sha256 -in {{ temp_test_folder.path }}/autorun.zip -out {{ temp_test_folder.path }}/autorun.zip.enc -pass file:{{ temp_test_folder.path }}/secret.key
```

Comment (rbuisson) [reply roles/usb_autorunner/tasks/main.yml]: (mind the `enc`too)

Comment (chinzou) [reply roles/usb_autorunner/tasks/main.yml]: Ah yes, sure.

**Pull request merged**